libraries/Wire/src/Wire.cpp

```diff
@@ -120,136 +133,28 @@ bool TwoWire::begin(int sdaPin, int sclPin, uint32_t frequency)
 void TwoWire::setTimeOut(uint16_t timeOutMillis)
 {
     _timeOutMillis = timeOutMillis;
+    i2cSetTimeOut(i2c, timeOutMillis);
```

Comment:
According to the [IDF documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/i2c.html#_CPPv415i2c_set_timeout10i2c_port_ti), the timeout is not in units of a msec, but its unit is: APB 80Mhz clock cycle.

Also it is a bit strange to have it here offered as millis, while the somewhat related stretch limit in the ESP8266 implementation is in usec.
I guess it will break some implementations out there that use the timeout, but at least the parameter suggestion is not what it does.
Having it in msec is fine I guess, but then you need to convert it in units of 1/80th MHz.  (thus effectively multiply it by 80000)

Reply:
@TD-er 

Thanks for the comment. 
TwoWire::setTimeOut is a API from Arduino Stream Class.
https://www.arduino.cc/reference/en/language/functions/communication/stream/streamsettimeout/
It takes its argument in milliseconds. Thus, the implementation follows Arduino standard.

If users follow Arduino Standards, nothing will break.

The link you sent is regarding something else, not the whole transaction timeout, but actually the time in cycles that i2c bus can be keept idle in order to the ESP i2c hardware machine state considers that there was no response for ACK/NACK.
 
The implentation of  Arduino HAL i2cSetTimeOut() calculates FreeRTOS ticks and keep this value to use in the IDF I2C APIs regarading a full i2c transaction, but not the bus idle time in clocks.

You can follow the code in order to see how it's used.

Reply:
In this case, I understand there is nothing to change reagarding the matter you presented.
Please let me know if you see diferently or there is something I am missing.
Thank you.